### ISSUE TITLE
WSL integrations is broken by downgrading Rancher Desktop

### ISSUE THREAD

**Issue by IsaSih** [rancher-sandbox/rancher-desktop#6061]
### Actual Behavior

Once we downgrade from the 1.11 patch candidate to 1.10 with WSL integrations enabled, wsl helper fails to set a kubeconfig integration for the distro

### Steps to Reproduce

1. Run Rancher Desktop 1.11 patch candidate (msi installer from https://github.com/rancher-sandbox/rancher-desktop/actions/runs/6962627070 ) , installing the build for all users in the system
2. Enable WSL integrations
3. Verify that kubeconfig is a file in the distro, not a symlink by running `ls -l ~/.kube/config`
4. Perform a factory reset 
5. Uninstall 1.11 patch candidate via Add or remove programs in `Apps and features`
6. Install Rancher Desktop 1.10  (msi installer , installing the build for all users in the system)
7. Once main window is displayed, try to enable WSL integrations via the UI

Observe that the option is not available for selection due to an error in WSL

### Result

From `integrations.log`:

```
Could not set up kubeconfig integration for Ubuntu: c [Error]: C:\Windows\system32\wsl.exe exited with code 11
    at ChildProcess.<anonymous> (C:\Program Files\Rancher Desktop\resources\app.asar\dist\app\background.js:1:9554)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  command: [
    'C:\\Windows\\system32\\wsl.exe',
    '--distribution',
    'Ubuntu',
    '--exec',
    '/mnt/c/Program Files/Rancher Desktop/resources/resources/linux/wsl-helper',
    'kubeconfig',
    '--enable=false'
  ],
  code: 11,
  [Symbol(child-process.command)]: 'C:\\Windows\\system32\\wsl.exe --distribution Ubuntu --exec /mnt/c/Program Files/Rancher Desktop/resources/resources/linux/wsl-helper kubeconfig --enable=false'
}
```

All Logs:
[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13482164/logs.zip)

<img width="768" alt="Screen Shot 2023-11-27 at 2 36 56 PM" src="https://github.com/rancher-sandbox/rancher-desktop/assets/70187881/84dfd49d-5e40-4047-bf78-bd65703daa18">

### Expected Behavior

WSL helper should be able to reconfigure the distro after a downgrade in Rancher Desktop with no errors.

### Rancher Desktop Version

1.10

### Rancher Desktop K8s Version

1.26.3

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 11

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

**Comment by IsaSih** [comment]
Tried shutting down Rancher Desktop, removing kubeconfig file from the distro and restarting Rancher Desktop after file was removed by the same error persists.

**Comment by gunamata** [comment]
I could reproduce on my Win 10 with below steps:

- install patch version
- Enable WSL Integration for my Ubuntu distro. Below are the logs at this state.

[patch-logs-after-wsl-integration.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13482334/patch-logs-after-wsl-integration.zip)

- Factory Reset
- Uninstall the App from `Program and Features`
- Install 1.11. At this point, I could not reproduce the error meaning I could enable WSL Integration for my Ubuntu distro . Below are the logs at this state.
 
[logs-after-downgrade-to-1.11.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13482379/logs-after-downgrade-to-1.11.zip)

- I proceeded further and downgraded to 1.10. I hit the error message at this point. Here's the error screenshot and logs at this state.

![image](https://github.com/rancher-sandbox/rancher-desktop/assets/59538726/10d58e8c-76a1-4fd7-aece-6e5150c0c05d)


[logs-after-downgrade-to-1.10.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13482436/logs-after-downgrade-to-1.10.zip)

**Comment by gaktive** [comment]
Should be fixed once #6119 lands so we'll need to verify that.
